### PR TITLE
add node-addon-api package for linux

### PR DESCRIPTION
**Pull request by enzalito**
I wrote this package for [node-addon-api](https://github.com/nodejs/node-addon-api/tree/main) as the project relies on their own build system (node-gyp) that I didn't want to use.  
It only supports Linux for now as the `.gyp` files on their repository seem to indicate that platform specific steps are needed for Windows and Macos.

### CODE REVIEW

File: packages/n/node-addon-api/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package("node-addon-api")
+
+    set_kind("library", {headeronly = true})
+    set_homepage("https://github.com/nodejs/node-addon-api")
+    set_description("Module for using Node-API from C++")
+    set_license("MIT")
+
+    add_configs("errors", {
```
Comment (waruqi) [inline packages/n/node-addon-api/xmake.lua:8]: please use single line and simplify description

File: packages/n/node-addon-api/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        description = "Control the availability of the deprecated APIs.",
+        default = true,
+        type = "boolean"
+    })
+
+    set_urls("https://github.com/nodejs/node-addon-api/archive/refs/tags/$(version).tar.gz",
+        "https://github.com/nodejs/node-addon-api.git")
+    add_versions("v8.0.0", "42424c5206b9d67b41af4fcff5d6e3cb22074168035a03b8467852938a281d47")
+
+    add_deps("node", {kind = 'binary', system = true})
```
Comment (waruqi) [inline packages/n/node-addon-api/xmake.lua:24]: no this package in xmake-repo now.

File: packages/n/node-addon-api/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if errors == "except" then
+            package:add("defines", "NAPI_CPP_EXCEPTIONS")
+        end
+    end)
+
+    on_install("linux" ,function(package)
+        local include_dir = path.join(package:installdir(), "include")
+        os.mkdir(include_dir)
+        os.cp("*.h", include_dir)
+    end)
```
Comment (waruqi) [inline packages/n/node-addon-api/xmake.lua:49]: no on_test

File: packages/n/node-addon-api/xmake.lua (hunk 2 above)
Comment (enzalito) [reply packages/n/node-addon-api/xmake.lua]: I couldn't find it, removing `system = true` gives me this error:
```
error: package(node) not found!
```

Comment (waruqi) [reply packages/n/node-addon-api/xmake.lua]: so you should add node as package first. and add `on_fetch` to find system node in node package at least.

File: packages/n/node-addon-api/xmake.lua (hunk 3 above)
Comment (waruqi) [reply packages/n/node-addon-api/xmake.lua:43]: please use package:has_cfuncs

**Comment by enzalito** [comment]
@waruqi could you merge this PR too if it looks good to you ?

File: packages/n/node-addon-api/xmake.lua (hunk 3 above)
Comment (enzalito) [reply packages/n/node-addon-api/xmake.lua:43]: I didn't see your comment my bad, I'll update the test

**Comment by enzalito** [comment]
@waruqi is the PR ready to be merged ?

**Pull request merged**